contributors/design-proposals/core-metrics-pipeline.md

```diff
@@ -0,0 +1,280 @@
+
+Many kubernetes system components currently support these features.  Many more components that support these features are in development.
+The following is not meant to be an exhaustive list, but gives the current set of use cases for these metrics.
+
+Metrics requirements for "First Class Resource Isolation and Utilization Features", based on kubernetes component needs, are as follows:  
+
+ - Kubelet
+   - Node-level availability metrics for Disk, Memory, and CPU  
+   - Pod-level usage metrics for Disk and Memory  
+ - Metrics Server (outlined in [Monitoring Architecture](https://github.com/kubernetes/kubernetes/blob/master/docs/design/monitoring_architecture.md)), which exposes the [Resource Metrics API](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/resource-metrics-api.md) to the following system components:
+   - Scheduler  
+     - Node-level capacity and availability metrics for Disk, CPU, and Memory  
+     - Pod-level usage metrics for Disk, CPU, and Memory  
+     - Container-level usage metrics for Disk, CPU, and Memory  
+   - Horizontal-Pod-Autoscaler  
+     - Node-level capacity and availability metrics for CPU and Memory  
+     - Pod-level usage metrics for CPU and Memory  
+   - Cluster Federation  
+     - Node-level capacity and availability metrics for Disk, Memory, and CPU  
+   - kubectl top and Kubernetes Dashboard  
+     - Node-level capacity and availability metrics for Disk, Memory, and CPU  
+     - Pod-level usage metrics for Disk, Memory, and CPU  
+     - Container-level usage metrics for Disk, CPU, and Memory  
+
+### Proposed Core Metrics API:
+
+An important difference between the current summary api and the proposed core metrics api is that pod and container-level metrics in the core metrics api contain only usage data, and not capacity-related statistics.  Node level resource metrics continue to provide capacity.  This is more accurate since a pod's resource capacity is really defined by its "requests" and "limits", and it is a better reflection of how components use these metrics.  The kubelet, for example, finds which resources are constrained using node-level capacity and availability data, and then chooses which pods to take action on based on the pod's usage of the constrained resource.  If neccessary, capacity for resources a pod consumes can still be correlated with node-level resources using this format of metrics.
+
+```go
+// CoreMetrics is a top-level container for holding NodeResources and PodUsage.  
+type CoreMetrics struct {  
+  // Overall node resource metrics.  
+  Node NodeResources   
+  // Per-pod usage metrics.  
+  Pods []PodUsage  
+}  
+
+// NodeResources holds node-level metrics.  NodeResources contains capacity and availibility for Node Resources.  
+type NodeResources struct {   
+  // The filesystem used by node kubernetes components.  
+  KubeletFilesystem string 
```

Comment:
Nit: I'd refer to this as the `PrimaryFilesystem` instead to align with upcoming support for local storage. Kubelet happens to consume the `PrimaryFilesystem` and is not the only consumer.